src/Core/Services/Implementations/NotificationsApiPushNotificationService.cs

```diff
@@ -197,7 +197,7 @@ private async Task SendMessageAsync<T>(PushType type, T payload, bool excludeCur
 
     private string GetContextIdentifier(bool excludeCurrentContext)
     {
-        if (!excludeCurrentContext)
+        if (excludeCurrentContext)
```

Comment:
I see that here you've reversed the condition here.  (I assume thisi was a bug?)

I think we also need to reverse the condition in the other implementations as well.  I noticed it's still backwards in `AzureQueuePushNotificationService` and `NotificationHubPushNotificationService`.

There's a slightly different set of logic in `RelayPushNotificationService` but it also appears to be backwards (i.e. "excludes == true" will actually include the DeviceIdentifier).

Reply:
This is vestigial from my misunderstanding based on the name.

It is meant to refer to exclude the currently authed client from this push notification -- something that requires adding the contextId to the notification.

I opted to change the call to `PushUserLogout` rather than invert the logic because it results in far fewer behavioral changes. This changeset (now that this mistake is reverted) specifically inverts logic only for password update requests, where we trust the requesting client to log out in order to provide them enough time to process the cipher-text updates required of a key rotation.